src/streamlink/plugins/ceskatelevize.py

```diff
@@ -5,131 +5,178 @@
+            "https://playlist.ceskatelevize.cz/",
+            data={
+                "data": json.dumps(
+                    {
+                        "contentType": "live",
+                        "items": [{
+                            "id": video_id,
+                            "key": f"{key}",
+                            "assetId": "CT4DRM",
+                            "playerType": "dash",
+                            "date": f"{date}",
+                            "requestSource": "front-sport",
+                            "drm": 0,
+                            "quality": "web",
+                        }],
+                    },
+                    separators=(',', ':'),
+                ),
+            },
             schema=validate.Schema(
                 validate.parse_json(),
                 {
-                    "playlist": [{
-                        validate.optional("type"): str,
-                        "streamUrls": {
-                            "main": validate.url(),
-                        }
-                    }]
+                    "RESULT": self.schema_playlist,
                 },
-                validate.get(("playlist", 0, "streamUrls", "main"))
-            )
+                validate.get(("RESULT", "playlist", 0, "streamUrls", "main")),
+            ),
         )
-        return DASHStream.parse_manifest(self.session, url)
+
+    def get_channel(self):
+        data = validate.Schema(
+            validate.parse_html(),
+            validate.xml_xpath_string(".//script[@id='__NEXT_DATA__'][text()]/text()"),
+            str,
+            validate.parse_json(),
+            {
+                "props": {
+                    "pageProps": {
+                        validate.optional("data"): {
+                            "liveBroadcast": {
+                                "id": str,
+                                "current": validate.none_or_all(
+                                    {
+                                        "channelName": str,
+                                    },
+                                ),
+                            },
+                        },
+                    },
+                },
+            },
+            validate.get(("props", "pageProps")),
+        ).validate(self.res.content)
+
+        if not data:
+            return
+
+        data = data.get("data").get("liveBroadcast")
+        self.id = data.get("id")
+
+        data = data.get("current")
+        if not data:
+            return
+
+        self.title = data.get("channelName")
+
+        return self.get_stream_url(self.id)
+
+    def _get_streams(self):
+        self.res = self.session.http.get(self.url)
+
+        if "://ct24" in self.res.url:
+            url = self.get_ct24()
+        elif "://decko" in self.res.url:
+            url = self.get_decko()
+        elif "://sport" in self.res.url:
+            url = self.get_sport()
+        else:
+            url = self.get_channel()
```

Comment:
As said, it's better to check via the `self.matches` list, because it's independent of the actual URLs. The only downside (right now) is the numeric indexing, but that's not too bad.

Or is there a specific reason to check the response URL due to redirections?